problem_builder/table.py

```diff
@@ -66,23 +75,148 @@ class MentoringTableBlock(StudioEditableXBlockMixin, StudioContainerXBlockMixin,
             {"display_name": "Immunity Map", "value": "immunity-map"},
         ],
     )
-    editable_fields = ("type", )
+    editable_fields = ("type", "allow_download")
+    allow_download = Boolean(
+        display_name=_("Allow Download"),
+        help=_("Allow students to download a copy of the table for themselves."),
+        default=False,
+        scope=Scope.content
+    )
+    allow_sharing = Boolean(
+        display_name=_("Allow Sharing"),
+        help=_("Allow students to share their results with other students."),
+        default=True,
+        scope=Scope.content
+    )
+    # Stores who a user has allowed to see their table.
+    shared_with = Dict(scope=Scope.user_state_summary)
+    # Stores whose tables a user is allowed to view. This needs to be kept in sync with the previous field.
+    view_options = Dict(scope=Scope.user_state_summary)
```

Comment:
This might need to be changed :-/

`Scope.user_state_summary` fields have no locking so if two students make changes at the same time, the second one will overwrite the first one. It also could easily lead to these two fields being out of sync, if a different student "wins" for each write.

Quote from Dave when I proposed something similar: "I would strongly discourage you from using Scope.user_state_summary for this purpose. As you point out, that scope was not meant to be used for items with high concurrency needs, and write conflicts will happen."